src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -918,12 +918,19 @@ class ComponentType:
         :param visibility: whether this member is public or private.
         :param initial_value: the initial value of this member (if any).
         :type initial_value: Optional[:py:class:`psyclone.psyir.nodes.Node`]
+        :param preceding_comment: a comment that precedes this component.
+        :type preceding_comment: Optional[str]
+        :param inline_comment: a comment that follows this component on the
+                               same line.
+        :type inline_comment: Optional[str]
         '''
         name: str
         # Use Union for compatibility with Python < 3.10
         datatype: Union[DataType, DataTypeSymbol]
         visibility: Symbol.Visibility
         initial_value: Any
+        preceding_comment: str = ""
+        inline_comment: str = ""
```

Comment:
In fact, I see that we still have code (and comments; mea culpa) that assumes that ComponentType is an immutable namedtuple:
https://github.com/stfc/PSyclone/blob/2d1f95338e9f34798fba686e6e6064e987c25ef4/src/psyclone/psyir/symbols/datatypes.py#L1059-L1077
If you 'unfreeze' ComponentType then presumably this could be simplified :-)

Reply:
Thanks @arporter , I think it is a good practise to make them frozen/immutable when possible (there is less moving parts to consider as possible errors), and just make them mutable when needed (maybe here?). @JulienRemy is this ready for another review? I could have a look at this before doing more changes.

Reply:
Thanks @arporter and @sergisiso. Let's leave it frozen then :)

This in indeed ready for another review, unless you'd rather I solved the merge conflicts first?

Reply:
It's just missing the "Ready for review" label :)

(but it will be great if you also solve the merge conflicts first)

Reply:
@sergisiso Merged and labelled :)